commit 554cb97172454939e905fcc4d50e4b078eb3d19c (refs/changes/86/2486/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-07T11:30:24-07:00 (2 years, 1 month ago)
    
    NET-376 hn-netfile should work for a server that isn't yet a headnode
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/Makefile b/Makefile
index 598cb40..dafc197 100644
--- a/Makefile
+++ b/Makefile
@@ -44,7 +44,7 @@ CLEAN_FILES += ./node_modules build/pkg dhcpd-pkg-*.tar.bz2
 REPO_MODULES := src/node-pack
 JSSTYLE_FLAGS = -o indent=4,doxygen,unparenthesized-return=0
 
-NODE_PREBUILT_VERSION=v0.10.32
+NODE_PREBUILT_VERSION=v0.10.48
 ifeq ($(shell uname -s),SunOS)
 	NODE_PREBUILT_TAG=zone
 	NODE_PREBUILT_IMAGE=de411e86-548d-11e4-a4b7-3bb60478632a
diff --git a/bin/hn-netfile b/bin/hn-netfile
index 9931490..c0e711c 100755
--- a/bin/hn-netfile
+++ b/bin/hn-netfile
@@ -118,37 +118,37 @@ function getHeadnodeInfo(opts, callback) {
 
     var i;
 
-    opts.cnapi.listServers({headnode: true}, function (err, headnodes) {
+    opts.cnapi.listServers(function (err, servers) {
         if (err) {
             callback(err);
             return;
         }
 
         // Try `uuid` match first ...
-        var headnode;
-        for (i = 0; i < headnodes.length; i++) {
-            if (headnodes[i].uuid === opts.headnodeHostnameOrUuid) {
-                headnode = headnodes[i];
+        var server;
+        for (i = 0; i < servers.length; i++) {
+            if (servers[i].uuid === opts.headnodeHostnameOrUuid) {
+                server = servers[i];
                 break;
             }
         }
 
         // ... then `hostname`.
-        if (!headnode) {
-            for (i = 0; i < headnodes.length; i++) {
-                if (headnodes[i].hostname === opts.headnodeHostnameOrUuid) {
-                    headnode = headnodes[i];
+        if (!server) {
+            for (i = 0; i < servers.length; i++) {
+                if (servers[i].hostname === opts.headnodeHostnameOrUuid) {
+                    server = servers[i];
                     break;
                 }
             }
         }
 
-        if (headnode) {
-            opts.cn_uuid = headnode.uuid;
-            opts.hostname = headnode.hostname;
+        if (server) {
+            opts.cn_uuid = server.uuid;
+            opts.hostname = server.hostname;
             callback();
         } else {
-            callback(new Error('Could not find a headnode=true server "'
+            callback(new Error('Could not find server "'
                 + opts.headnodeHostnameOrUuid + '" in CNAPI'));
         }
     });
diff --git a/package.json b/package.json
index de942c1..464afdb 100644
--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "mkdirp": "0.5.0",
     "node-uuid": "1.2.0",
     "restify": "2.8.4",
-    "sdc-clients": "git+https://github.com/joyent/node-sdc-clients.git#0aea3cf",
+    "sdc-clients": "10.3.0",
     "sdc-bunyan-serializers": "git+https://github.com/joyent/sdc-bunyan-serializers.git#5454ae4",
     "sprintf": "0.1.0",
     "vasync": "1.6.2"
@@ -20,7 +20,7 @@
   "devDependencies": {
     "clone": "0.1.4",
     "mockery": "1.2.0",
-    "nodeunit": "0.7.4"
+    "nodeunit": "0.11.1"
   },
   "scripts": {
     "test": "make test"
